pulseguardian/guardian.py

```diff
@@ -56,8 +57,32 @@ def clear_deleted_queues(self, queues):
         for queue in deleted_queues:
             logging.info("Queue '{0}' has been deleted.".format(queue))
             db_session.delete(queue)
+
+        # Clean up bindings on queues that are not deleted.
+        for queue_name in alive_queues_names:
+            self.clear_deleted_bindings(queue_name)
+
         db_session.commit()
 
+    def clear_deleted_bindings(self, queue_name):
+        rabbit_bindings = pulse_management.queue_bindings(config.rabbit_vhost,
+                                                          queue_name)
+
+        db_bindings = Binding.query.filter(Binding.queue_name == queue_name)
+
+        # Filter bindings that are in the database but no longer on RabbitMQ.
+        alive_bindings_names = {Binding.as_string(b['source'], b['routing_key'])
+                                for b in rabbit_bindings}
+
+        deleted_bindings = set(b for b in db_bindings if b.name
```

Comment:
Oops.  Copy/paste leftover from ``clear_deleted_queues``.   Fixed it there, too.